src/test/suite/didactUriCompletionItemProvider.test.ts

```diff
@@ -215,3 +265,55 @@ async function checkForCommandInList(completions: vscode.CompletionItem[], label
 	}
 	return false;
 }
+
+function delay(ms: number) {
+	return new Promise( resolve => setTimeout(resolve, ms) );
+}
+
+async function executeCompletionTest(input: string, expected: string, selectLastSuggestion? : boolean) {
+	const testFolder = path.resolve(__dirname, '..', '..', '..', './test Fixture with speci@l chars');
+	const filename = path.resolve(testFolder, 'testmy.didact.md');
+	const uri = vscode.Uri.parse(filename);
+	await vscode.workspace.fs.writeFile(uri, Buffer.from(input));
+	const document = await vscode.workspace.openTextDocument(uri);
+	const editor = await vscode.window.showTextDocument(document, vscode.ViewColumn.One, true);
+	await delay(500);
+
+	const newCursorPosition = new vscode.Position(0, input.length);
+	editor.selection = new vscode.Selection(newCursorPosition, newCursorPosition);
+
+	const actualCompletionList = (await vscode.commands.executeCommand(
+		'vscode.executeCompletionItemProvider',
+		document.uri, newCursorPosition)) as vscode.CompletionList;
+	expect(actualCompletionList).to.not.be.null;
+	expect(actualCompletionList.items.length).to.be.at.least(1);
+
+	// if this is available, we have completed the didact://?commandId= part of the completion
+	const startCompletionExists = await checkForCommandInList(actualCompletionList.items, "Start new Didact command link");
+
+	// if this is available we have populated the command list
+	const startCommandCompletionExists = await checkForCommandInList(actualCompletionList.items, "vscode.didact.startDidact");
+
+	// if either is complete, we have expected completions showing up
+	try {
+		expect(startCompletionExists).to.be.true;
+	} catch (e) {
+		expect(startCommandCompletionExists ).to.be.true;		
+	}
+
+	await vscode.commands.executeCommand("editor.action.triggerSuggest");
+	await delay(1000);
+
+	// bump the selection down to the last suggestion
+	if (selectLastSuggestion) {
+		await vscode.commands.executeCommand("selectLastSuggestion");
+		await delay(500);
+	}
+	await vscode.commands.executeCommand("acceptSelectedSuggestion");
+	await delay(1000);
+	await vscode.commands.executeCommand('editor.action.selectAll');
+	expect(editor.document.getText()).to.be.equal(expected);
+
+	await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+	await vscode.workspace.fs.delete(uri);
```

Comment:
it sounds like this should be in the afterEach so that in case of test failure, it doesn't impact other tests

Reply:
Only problem with that is the uri is generated in the executeCompletionTest function, not in the suite where I have access to the afterEach. Not sure I want to split out creating the file from the test

Reply:
Tried removing it in favor of just having the removeFilesAndFolders function handle it in the afterEach and it wasn't enough to get the tests to complete reliably. So I have left in the delete for now.